src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -183,6 +217,28 @@ private void applyStringFilter(String filterString) {
     }
 
     /**
+     * Triggered when a RepoOpeningEvent is received by the filter panel
+     *
+     */
+    private void addPanelLoadingIndication() {
+        logger.info("Preparing to add panel loading indication");
+        synchronized (this) {
```

Comment:
Generally we try to avoid this sort of low-level synchronisation, as it is ad-hoc and doesn't compose well.

In this case Platform.runLater should be the solution. You have to either make sure these events are always triggered from the UI thread (like @acruis suggested), or received on the UI thread. I don't think there's a big difference here, but you should use the one that leads to less duplication (so we can't forget to synchronise somewhere).